### ISSUE TITLE
How do we let users disable recursive imports?

### ISSUE THREAD

**Issue by oconnor663** [buildinspace/peru#123]
In particular, if you're importing a module that *checks in* its own imports, you will get sync failures even without any override voodoo. Right now the way to disable recursive imports is `peru file: NONEXISTENT_NAME`, which is not elegant.

**Comment by oryband** [comment]
what is this `peru file: NONEXISTENT_NAEM` ? where do you use it? in the command line? in `peru.yaml`? can you elaborate?

**Comment by oconnor663** [comment]
Apologies for the lack of detail above. `peru file` is a field that you can include in any module. So for example, you could define:

```yaml
git module my_project:
  url: https://github.com/me/my_project
  peru file: my_peru_file.yaml
```

That would cause peru to read `my_peru_file.yaml` from the root of `my_project` to determine the recursive imports, rather than looking for the usual `peru.yaml` file in there. As a special case of this, you can pass a nonexistent file name here to disable recursive imports entirely.

I'm not a big fan of this design, which is one of the reasons it's not documented yet. The most common use case is the disable-imports-entirely hack, so the design feels backwards. One issue is that so far we've managed to have every field be of the string type, but what we need here looks more like a boolean flag -- something along the lines of `recursive imports: false` -- which requires some more thinking. (Should boolean values silently convert to strings when a string is expected, or should that be an error? Etc.)

**Comment by oconnor663** [comment]
Implemented in c10c3681b91a18e6bfa6e341cc3f85e0d2f266b3, as part of https://github.com/buildinspace/peru/issues/146. The new `recursive` field currently defaults to `true` with a warning, but the thinking on that issue is that we're going to start disabling recursive modules by default.
